python/dgl/graphbolt/itemset.py

```diff
@@ -442,3 +443,188 @@ def __repr__(self) -> str:
             itemsets=itemsets_str,
             names=self._names,
         )
+
+
+class ItemSet4(Dataset):
+    r"""Class for iterating over tensor-like data.
+    Experimental. Implemented only __getitem__() accepting slice and list.
+    """
+
+    def __init__(
+        self,
+        items: Union[torch.Tensor, Mapping, Tuple[Mapping]],
```

Comment:
do we need `Mapping`? we do need `int`, right?

Reply:
1. I'm considering to disable `int`, because `int` doesn't contain any `dtype` info. To generate an `all_nodes_set` we can always use a tensor scalar (which is how we do it now). What's your opinion?
2. `Mapping` is indded to "large-scope", but I'm trying to differentiate `ItemSet4` from existing `ItemSet` which takes in `Iterable`. I think `Sequence` might be a good choice?

Reply:
https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence
`Sequence` best matches our design in my opinion.